src/sentry/taskworker/worker.py

```diff
@@ -311,11 +303,20 @@ def shutdown(self) -> None:
         if self._shutdown_event.is_set():
             return
 
+        logger.info("taskworker.worker.shutdown")
         self._shutdown_event.set()
+        if self._result_thread:
+            self._result_thread.join()
+
+        # Drain remaining results synchronously, as the thread will have terminated
+        # when shutdown_event was set.
         while True:
-            more = self._drain_result(fetch=False)
-            if not more:
+            try:
+                result = self._processed_tasks.get_nowait()
+            except queue.Empty:
                 break
+            self._send_result(result, fetch=False)
+
         for child in self._children:
```

Comment:
Shouldn't we actually drain child pool first and ensure the child is empty before we synchronously drain the remaining results queue?

Reply:
Good point, joining the children could produce more results.